### PR TITLE
many: pass consistently boot.Device state to boot methods

### PR DESCRIPTION
**Pull request by pedronis**
instead of passing model and/or a classic flag to them
and adding operating mode now as well, pass a new interface
boot.Device that is implemented by DeviceContext

### CODE REVIEW

**Comment by pedronis** [comment]
based on #7898

File: boot/boot.go [hunk 1]
```diff
@@ -73,12 +73,15 @@ var _ BootParticipant = trivial{}
 // ensure trivial is a Kernel
 var _ BootKernel = trivial{}
 
-// Model carries information about the model that is relevant to boot.
-// Note *asserts.Model implements this, and that's the expected use case.
-type Model interface {
+// Device carries information about the devie model and mode that is
+// relevant to boot.  Note snapstate.DeviceContext implements this, and that's
```
Comment (bboozzoo) [inline boot/boot.go]:
```suggestion
// relevant to boot. Note snapstate.DeviceContext implements this, and that's
```

File: boot/boottest/device.go [hunk 2]
```diff
@@ -0,0 +1,45 @@
...
+ *
+ */
+
+package boottest
+
+import (
+	"strings"
+)
+
+// MockDevice implements boot.Device. It wraps a string like
```
Comment (mvo5) [inline boot/boottest/device.go]: This feels a bit magic but it's fine, we can trivial change it later if it becomes a problem.

File: overlord/snapstate/handlers.go [hunk 3]
```diff
@@ -1184,9 +1195,7 @@ func (m *SnapManager) doLinkSnap(t *state.Task, _ *tomb.Tomb) (err error) {
 		return err
 	}
 
-	// XXX: this block is slightly ugly, find a pattern when we have more examples
```
Comment (mvo5) [inline overlord/snapstate/handlers.go]: Nice! One XXX down

**Pull request merged**